test/cmdlineargs.jl

```diff
@@ -0,0 +1,168 @@
+let exename = joinpath(JULIA_HOME, (ccall(:jl_is_debugbuild, Cint, ()) == 0 ? "julia" : "julia-debug"))
+    # --version
+    let v = split(readall(`$exename -v`), "julia version ")[end]
+        @test VERSION == VersionNumber(v)
+    end
+    @test readall(`$exename -v`) == readall(`$exename --version`)
+
+    # --help
+    @test startswith(readall(`$exename -h`), "julia [options] [program] [args...]")
+    @test startswith(readall(`$exename --help`), "julia [options] [program] [args...]")
+
+    # --quiet
+    # This flag is indirectly tested in test/repl.jl
+
+    # --home
+    @test success(`$exename -H $JULIA_HOME`)
+    @test success(`$exename --home=$JULIA_HOME`)
+
+    # --eval
+    @test  success(`$exename -e "exit(0)"`)
+    @test !success(`$exename -e "exit(1)"`)
+    @test  success(`$exename --eval="exit(0)"`)
+    @test !success(`$exename --eval="exit(1)"`)
+    @test !success(`$exename -e`)
+    @test !success(`$exename --eval`)
+
+    # --print
+    @test readall(`$exename -E "1+1"`) == "2\n"
+    @test readall(`$exename --print="1+1"`) == "2\n"
+    @test !success(`$exename -E`)
+    @test !success(`$exename --print`)
+
+    # --post-boot
+    @test  success(`$exename -P "exit(0)"`)
+    @test !success(`$exename -P "exit(1)"`)
+    @test  success(`$exename --post-boot="exit(0)"`)
+    @test !success(`$exename --post-boot="exit(1)"`)
+    @test !success(`$exename -P`)
+    @test !success(`$exename --post-boot`)
+
+    # --load
+    let testfile = tempname()
+        try
+            open(testfile, "w") do io
+                println(io, "testvar = :test")
+            end
+            @test split(readchomp(`$exename --load=$testfile -P "println(testvar)"`), '\n')[end] == "test"
+            @test split(readchomp(`$exename -P "println(testvar)" -L $testfile`), '\n')[end] == "test"
+        finally
+            rm(testfile)
+        end
+    end
+    # -L, --load requires an argument
+    @test !success(`$exename -L`)
+    @test !success(`$exename --load`)
+
+    # --cpu-target
+    @test !success(`$exename -C invalidtarget`)
```

Comment:
We could turn the test off, I don't really care.  We do fail here with the sysimg but I don't know why anyone would rely on that behavior.